libs/CovidTimeseriesModelSIR.py

```diff
@@ -243,93 +248,93 @@ def iterate_model(self, model_parameters):
 
-            actuals.index = pd.to_datetime(actuals.index, format="%Y-%m-%d")
+        actuals["infected_a"] = 0
+        actuals["infected_b"] = 0
+        actuals["infected_c"] = 0
 
-            actuals["infected_a"] = 0
-            actuals["infected_b"] = 0
-            actuals["infected_c"] = 0
+        actuals["exposed"] = 0
 
-            actuals["exposed"] = 0
+        all_cols = [
+            "total",
+            "susceptible",
+            "exposed",
+            "infected",
+            "infected_a",
+            "infected_b",
+            "infected_c",
+            "recovered",
+            "dead",
+        ]
 
-            all_cols = [
-                "total",
-                "susceptible",
-                "exposed",
-                "infected",
-                "infected_a",
-                "infected_b",
-                "infected_c",
-                "recovered",
-                "dead",
-            ]
+        actuals["susceptible"] = 0
+        sir_df["susceptible"] = 0
 
-            actuals["susceptible"] = 0
-            sir_df["susceptible"] = 0
+        actuals = actuals.loc[:, all_cols]
+        sir_df = sir_df.loc[:, all_cols]
 
-            actuals = actuals.loc[:, all_cols]
-            sir_df = sir_df.loc[:, all_cols]
+        combined_df = pd.concat([actuals, sir_df])
 
-            combined_df = pd.concat([actuals, sir_df])
+        if model_parameters["interventions"] is not None:
+            (combined_df, counterfactuals) = self.run_interventions(
+                model_parameters, combined_df, init_params, model_seir_init, r0
+            )
 
-            if model_parameters["interventions"] is not None:
-                (combined_df, counterfactuals) = self.run_interventions(
-                    model_parameters, combined_df, init_params, r0
-                )
+        # this should be done, but belt and suspenders for the diffs()
+        combined_df.sort_index(inplace=True)
+        combined_df.index.name = "date"
+        combined_df.reset_index(inplace=True)
 
-            # this should be done, but belt and suspenders for the diffs()
-            combined_df.sort_index(inplace=True)
-            combined_df.index.name = "date"
-            combined_df.reset_index(inplace=True)
+        combined_df["total"] = pop_dict["total"]
 
-            combined_df["total"] = pop_dict["total"]
+        # move the actual infected numbers into infected_a where its NA
+        combined_df["infected_a"] = combined_df["infected_a"].fillna(
+            combined_df["infected"]
+        )
 
-            # move the actual infected numbers into infected_a where its NA
-            combined_df["infected_a"] = combined_df["infected_a"].fillna(
-                combined_df["infected"]
+        if model_parameters["model"] == "seir":
+
+            # make infected total represent the sum of the infected stocks
+            # but don't overwrite the historical
+            combined_df.loc[:, "infected_tmp"] = (
```

Comment:
I don't think you need to add this to the dataframe:

```
infected_tmp = (...)
combined_df.loc[:, "infected"].fillna(infected_tmp)
```
should work!

Reply:
This infected sum does change for each row... the thing I'm trying to achieve is fill infected for the modeled rows without overwriting infected in the historical rows where the _a _b _c stocks are empty (thus causing the historical to read 0). Fillna from the column is clunky, but the best way I could think to achieve that.